### ISSUE TITLE
[BUG] Error "invalid use of incomplete type 'class LAMMPS_NS::Atom'"

### ISSUE THREAD

**Issue by baip** [lammps/lammps#2669]
**Summary**

Error "invalid use of incomplete type 'class LAMMPS_NS::Atom'" while building LAMMPS, similar as [this issue](https://github.com/lammps/lammps/issues/1720).

**LAMMPS Version and Platform**

LAMMPS commit 42895a
Ubuntu 20.04 LTS
GCC 9.3.0 + MPICH2 + FFTW3 from Conda:
~~~
binutils_impl_linux-64    2.35.1               h193b22a_2    conda-forge
binutils_linux-64         2.35                h67ddf6f_30    conda-forge
fftw                      3.3.9           nompi_h74d3f13_101    conda-forge
gcc_impl_linux-64         9.3.0               h70c0ae5_18    conda-forge
gcc_linux-64              9.3.0               hf25ea35_30    conda-forge
gfortran_impl_linux-64    9.3.0               hc4a2995_18    conda-forge
gfortran_linux-64         9.3.0               hdc58fab_30    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
gxx_impl_linux-64         9.3.0               hd87eabc_18    conda-forge
gxx_linux-64              9.3.0               h3fbe746_30    conda-forge
mpc                       1.1.0             h04dde30_1009    conda-forge
mpfr                      4.0.2                he80fd80_1    conda-forge
mpi                       1.0                       mpich    conda-forge
mpi4py                    3.0.3            py38he865349_5    conda-forge
mpich                     3.4.1              h846660c_104    conda-forge
mpich-mpicc               3.4.1              h2cc385e_104    conda-forge
~~~

**Actual Behavior**

~~~
src/KSPACE/ewald.cpp: In member function 'virtual void LAMMPS_NS::Ewald::init()':
src/KSPACE/ewald.cpp:100:12: error: invalid use of incomplete type 'class LAMMPS_NS::Atom'
  100 |   if (!atom->q_flag) error->all(FLERR,"Kspace style requires atom attribute q");
      |            ^~
In file included from src/pointers.h:29,
                 from src/kspace.h:17,
                 from src/KSPACE/ewald.h:23,
                 from src/KSPACE/ewald.cpp:21:
src/lammps.h:30:9: note: forward declaration of 'class LAMMPS_NS::Atom'
   30 |   class Atom *atom;              // atom-based quantities
      |         ^~~~
~~~

**Steps to Reproduce**

~~~
$ cmake -DLAMMPS_OMP_COMPAT=4 -DCMAKE_CXX_COMPILER=mpicxx -DCMAKE_C_COMPILER=mpicc -DCMAKE_Fortran_COMPILER=mpif90 -DCMAKE_CXX_FLAGS='-O3 -mtune=native -march=native' -DBUILD_MPI=yes -DBUILD_OMP=yes -DFFT=FFTW3 -DLAMMPS_MEMALIGN=64 -DBUILD_SHARED_LIBS=on -DLAMMPS_EXCEPTIONS=yes -DPKG_PYTHON=on -DFFTW3_INCLUDE_DIR=/share/apps/conda/envs/conda3.8/include -DFFTW3_LIBRARY=/share/apps/conda/envs/conda3.8/lib -DCMAKE_INSTALL_PREFIX=$HOME/tools/lammps/gcc-mpich2-fftw3 -C ../cmake/presets/minimal.cmake ../cmake
$ make
~~~

**Comment by akohlmey** [comment]
Sorry, but I cannot reproduce this.

The error message does not make sense, because the file "src/KSPACE/ewald.cpp" **does** include the necessary header "atom.h".

This would mean that either you have a modified LAMMPS installation with some incorrectly modified files, or something forces including a file `atom.h` not from LAMMPS (but I would not know how this could be possible). I can use the same conda-forge packages that you list and I tested on a ubuntu 20.04 machine:

```
binutils_impl_linux-64    2.35.1               h193b22a_2    conda-forge
binutils_linux-64         2.35                h67ddf6f_30    conda-forge
fftw                      3.3.9           nompi_h74d3f13_101    conda-forge
gcc_impl_linux-64         9.3.0               h70c0ae5_18    conda-forge
gcc_linux-64              9.3.0               hf25ea35_30    conda-forge
gfortran_impl_linux-64    9.3.0               hc4a2995_18    conda-forge
gfortran_linux-64         9.3.0               hdc58fab_30    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
gxx_impl_linux-64         9.3.0               hd87eabc_18    conda-forge
gxx_linux-64              9.3.0               h3fbe746_30    conda-forge
mpc                       1.1.0             h04dde30_1009    conda-forge
mpfr                      4.0.2                he80fd80_1    conda-forge
mmpi                       1.0                       mpich    conda-forge
mpi4py                    3.0.3            py38he865349_5    conda-forge
mpich                     3.4.1              h846660c_104    conda-forge
mpich-mpicc               3.4.1              h2cc385e_104    conda-forge
```

**Comment by baip** [comment]
Thanks, Axel, for looking into this so quickly! Yes, I saw that "src/KSPACE/ewald.cpp" included "atom.h" and the LAMMPS source files were not modified at all. That's why it felt so weird. I also tested on two different Ubuntu 20.04 machines.

HOWEVER, I was able to compile successfully just now inside a brand new conda environment with only the listed packages (sorry I should have tried this first). It looks like that other Conda packages somehow managed to interfere with LAMMPS compilation, which based on the error messages I didn't think was relevant/possible.
